src/py21cmfast/io/h5.py

```diff
@@ -307,27 +309,35 @@ def read_inputs(
 
 def _read_inputs_pre_v4(group: h5py.Group, safe: bool = True):
     input_classes = [
-        istruct.UserParams,
+        istruct.MatterParams,
+        istruct.MatterFlags,
         istruct.CosmoParams,
         istruct.AstroParams,
-        istruct.FlagOptions,
+        istruct.AstroFlags,
     ]
-    input_class_names = [cls.__name__ for cls in input_classes]
 
-    # Read the input parameter dictionaries from file.
-    kwargs = {}
-    for k in attrs.fields_dict(InputParameters):
-        kfile = k.lstrip("_")
-        input_class_name = snake_to_camel(kfile)
-
-        if input_class_name in input_class_names:
-            kls = input_classes[input_class_names.index(input_class_name)]
+    old_names = [
+        "user_params",
+        "cosmo_params",
+        "astro_params",
+        "flag_options",
+        "global_params",
+    ]
 
-            subgrp = group[kfile]
-            dct = dict(subgrp.attrs)
-            kwargs[k] = kls.from_subdict(dct, safe=safe)
-        else:
-            kwargs[k] = group.attrs[kfile]
+    # Read the input parameter dictionaries from file.
```

Comment:
Do we have a test of whether this works? We should definitely have one before we release.

Reply:
Due to how much the parameters have changed I would expect this to not produce a complete parameter set most of the time. Parameter names in v3 which are not in v4 will be ignored and parameter names in v4 not in v3 will be set to defaults.

As far as verifying that behaviour, I guess I could hold v3 file somewhere for testing? I'm not sure how to do it otherwise